mage_ai/data_cleaner/cleaning_rules/remove_outliers.py

```diff
@@ -4,33 +4,58 @@
     ActionType,
     Axis,
 )
+from typing import Union, Dict
+import numpy as np
 
+ITREE_ANOMALY_SCORE_THRESHOLD = -0.10
+LOF_ANOMALY_SCORE_THRESHOLD = -1.5
 REMOVE_OUTLIERS_TITLE = 'Remove outliers'
 
 
 class RemoveOutliers(BaseRule):
-    def evaluate(self):
-        suggestions = []
-        for c in self.df_columns:
-            if f'{c}/outlier_count' not in self.statistics:
-                continue
-            outlier_count = self.statistics[f'{c}/outlier_count']
-            if outlier_count == 0:
-                continue
-            std = self.statistics[f'{c}/std']
-            avg = self.statistics[f'{c}/average']
+    """
+    Checks dataframe for the existence of potential outliers, and generates a cleaning suggestion
+    to remove these outliers
+    """
+
+    def __init__(self, df, column_types, statistics):
+        super().__init__(df, column_types, statistics)
+        self.numeric_df, self.numeric_columns = self._filter_numeric_types()
```

Comment:
no need to filter numeric columns here anymore. it takes additional time to filter numeric types.